.github/workflows/go-autoformat.yml

```diff
@@ -0,0 +1,48 @@
+name: go-format-tidy
+on:
+  push:
+    paths:
+      - '**/**.go'
+      - '**/go.mod'
+      - '.github/workflows/go-autoformat.yml'
+  pull_request:
+    paths:
+      - '**/**.go'
+      - '**/go.mod'
+      - '.github/workflows/go-autoformat.yml'
+jobs:
+  go-fmt-vet-tidy: #test and submit coverage to codecov on pushes
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-go@v2
+      with:
+        go-version: '^1.18' # The Go version to download (if necessary) and use.
+    - uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-autoformat-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-autoformat-
+    - name: git config
+      run: |
+        git config --global user.email "autoformat@oadp.io"
+        git config --global user.name "Auto Format"
+    - name: format
+      run: go fmt ./...
+    - name: commit & push changes # push step 1, show missing format before vet and tidy starts to provide early feedback
+      run: |
+        git add .
+        git commit -m "Auto Format: 'go fmt ./...'" && git push
```

Comment:
ya.. not ok.. imho.   
enforce the linting, code spec locally before it leaves a developer's laptop :)